righttyper/righttyper.py

```diff
@@ -243,8 +243,16 @@ def enter_function(code: CodeType, offset: int) -> Any:
             }
         else:
             defaults = {}
-
-        process_function_arguments(t, inspect.getargvalues(frame), defaults)
+        
+        self_type: type | None = None
+        args = inspect.getargvalues(frame).args
+        if args:
+            first_arg = frame.f_locals[args[0]]
+            for ancestor in first_arg.__class__.__mro__:
+                if function in ancestor.__dict__.values():
```

Comment:
I think you should be able to look for the function name directly, as in
```
if ancestor.__dict__.get(function.__name__, None) is function:
   ...
```

But you need automatic tests to validate this: with the class itself, as well as up/down the MRO.